### PR TITLE
Allow to configure that fluent setters are allowed for NoReturnSetterMethodRule

### PR DESCRIPTION
**Pull request by keulinho**

### CODE REVIEW

File: src/Rules/NoReturnSetterMethodRule.php [hunk 1]
```diff
@@ -63,15 +67,15 @@ public function processNode(Node $node, Scope $scope): array
         }
 
         $classMethodName = $node->name->toString();
-        if ($classMethodName === 'setUp') {
+        if (strpos($classMethodName, 'setUp') === 0) {
```
Comment (keulinho) [inline src/Rules/NoReturnSetterMethodRule.php:70]: IMHO `setUp` is it's own prefix that is not confused with a "normal" setter, so i think this rule should skip all methods prefixed with `setUp`

**Comment by TomasVotruba** [comment]
Hi, this would go against the rule idea, that tries to remove fluency.

Fluent interfaces could create ambiguous API. See https://ocramius.github.io/blog/fluent-interfaces-are-evil/

**Comment by keulinho** [comment]
I understand that fluency has also has a lot of downsides, that is why I made it configurable and disabled by default, so by default the behaviour of the rule does not change by default.

For me the rule does not only remove fluency, it is also a check for `Command Query Separation`, setting a value is a `command` that should not return a value. That's independent for me from the whole fluency discussion.

And on a side note other rules already allow to be configured to be used for fluent interfaces, e.g. the [NoChainMethodCallRule](https://github.com/symplify/phpstan-rules/blob/main/docs/rules_overview.md#nochainmethodcallrule)

**Comment by TomasVotruba** [comment]
That is quite specific condition, that is better to handle in your own rule.

**Pull request closed without merging**